lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -0,0 +1,310 @@
+# PostgreSQL module specific support methods.
+#
+def get_publication_info(cursor, publication):
+    query = """
+    SELECT pubname AS name, pubowner AS owner, puballtables AS alltables, pubinsert, pubupdate, pubdelete
+    FROM pg_publication
+    WHERE pubname = %(publication)s
+    """
+    cursor.execute(query, {'publication': publication})
+    cur_result = cursor.fetchone()
+    if cur_result is None:
+        return None
+
+    result = {
+        "name": cur_result['name'],
+        "owner": cur_result['owner'],
+        "alltables": cur_result['alltables'],
+        "insert": cur_result['pubinsert'],
+        "update": cur_result['pubupdate'],
+        "delete": cur_result['pubdelete'],
+        "tables": []
+    }
+
+    if result['alltables'] == 't':
+        return result
+
+    query = """
+    SELECT schemaname, tablename FROM pg_publication_tables WHERE pubname = %(publication)s
+    """
+    cursor.execute(query, {'publication': publication})
+    result['tables'] = cursor.fetchall()
+    return result
+
+
+def publication_exists(cursor, publication):
+    query = "SELECT pubname FROM pg_publication WHERE pubname = %(publication)s"
+    cursor.execute(query, {'publication': publication})
+    return cursor.rowcount == 1
+
+
+def publication_delete(cursor, publication):
+    if publication_exists(cursor, publication):
+        query = "DROP PUBLICATION %s" % pg_quote_identifier(publication, 'publication')
+        cursor.execute(query)
+        return True
+    else:
+        return False
+
+
+def publication_modify_tables(cursor, publication, tables):
+    if not publication_exists(cursor, publication):
+        raise AssertionError("Modifying inexistant publication.")
+
+    table_quoted_list = []
+    for table in tables:
+        table_quoted_list.append('%s' % pg_quote_identifier(table, 'table'))
+
+    cursor.execute("ALTER PUBLICATION %s SET TABLE %s" % (pg_quote_identifier(publication, 'publication'),
+                   ', '.join(table_quoted_list)))
+    return True
+
+
+def publication_create(cursor, publication, tables):
+    if not publication_exists(cursor, publication):
+        query_fragments = ['CREATE PUBLICATION %s FOR' % pg_quote_identifier(publication, 'publication')]
+        if tables:
+            table_quoted_list = []
+            for table in tables:
+                table_quoted_list.append('%s' % pg_quote_identifier(table, 'table'))
+                query_fragments.append('TABLE %s' % ', '.join(table_quoted_list))
+        else:
+            query_fragments.append('ALL TABLES')
+        query = ' '.join(query_fragments)
+        cursor.execute(query)
+        return True
+    else:
+        publication_info = get_publication_info(cursor, publication)
+        # If publication switch from targeted tables to all tables (or invert), recreate it
+        if publication_info['alltables'] is True and not tables or publication_info['alltables'] is True and not tables:
```

Comment:
This expression do look a bit suspcious, can you explain a bit more what you try to achieve ? (cause that's if "A and B or A and B" and I am a bit unsure on the order to which I need to evaluate things)